### PR TITLE
bug(customer, order, bigcommerce_api_error): fix errors

### PR DESCRIPTION
**Pull request by jhaltercmu2020**
fix errors

### CODE REVIEW

File: lib/client/customer.rb [hunk 1]
```diff
@@ -12,7 +12,12 @@ def get(id, params = {})
           return response.body['data'][0]
         rescue Faraday::Error::ClientError => e
           raise BigcommerceApiError.new(
-            'get order', "Failed to get order #{id}", { order_id: id }, e
+            500,
+            'get order customer',
+            "Failed to get order customer #{id}",
+            id,
+            { customer_id: id },
+            e
```
Comment (mrogers-5s) [inline lib/client/customer.rb]: @Jhalter5Stones The `status` value here should be coming from the faraday response, and we should be including any errors returned from BigCommerce (much like the product agent is doing.

Additionally, the process that generates error events probably needs to be updated to handle this modified error structure.

**Review by mrogers-5s — CHANGES_REQUESTED** [review]

File: lib/client/customer.rb (hunk 1 above)
Comment (jhaltercmu2020) [reply lib/client/customer.rb]: @mrogers-5s I think I resolved this but let me know what you think

File: lib/client/customer.rb [hunk 2]
```diff
@@ -12,7 +12,12 @@ def get(id, params = {})
           return response.body['data'][0]
         rescue Faraday::Error::ClientError => e
           raise BigcommerceApiError.new(
-            'get order', "Failed to get order #{id}", { order_id: id }, e
+            e.response[:status],
+            'get order customer',
+            "Failed to get order customer #{id}",
+            id,
+            { customer_id: id },
```
Comment (mrogers-5s) [inline lib/client/customer.rb]: @Jhalter5Stones This is redundant.

The first `id` should be the customer ID (as the identifier). The _data_ object should contain the order ID. And it should also include the errors from the BigCommerce response

File: lib/client/order.rb [hunk 3]
```diff
@@ -10,7 +10,14 @@ def get(id, params = {})
           response = client.get(uri({ order_id: id }), params)
           order = response.body
         rescue Faraday::Error => e
-          raise BigcommerceApiError.new('get order', { order_id: id }, e)
+          raise BigcommerceApiError.new(
+            e.response[:status],
+            'get order',
+            "Failed to get order #{id}",
+            id,
+            { order_id: id },
```
Comment (mrogers-5s) [inline lib/client/order.rb]: We're already passing the order ID. We don't need to include it in the data as well. This data object should only have the errors from the BigCommerce response

File: lib/client/order.rb [hunk 4]
```diff
@@ -34,7 +41,14 @@ def get_products(order_id, params = {})
           response = client.get(uri({ order_id: order_id }, 'products'), params)
           return response.body
         rescue Faraday::Error => e
-          raise BigcommerceApiError.new('get order products', { order_id: order_id }, e)
+          raise BigcommerceApiError.new(
+            e.response[:status],
+            'get order products',
+            "Failed to get order products #{order_id}",
+            order_id,
+            { order_id: order_id },
```
Comment (mrogers-5s) [inline lib/client/order.rb]: Same here. Drop `order_id` from the data, and make sure to add the errors from BC

File: lib/client/order.rb [hunk 5]
```diff
@@ -44,7 +58,14 @@ def get_shipping_addresses(order_id, params = {})
           response = client.get(uri({ order_id: order_id }, 'shipping_addresses'), params)
           return response.body
         rescue Faraday::Error => e
-          raise BigcommerceApiError.new('get order shipping address', { order_id: order_id }, e)
+          raise BigcommerceApiError.new(
+            e.response[:status],
+            'get order shipping address',
+            "Failed to get order shipping address #{order_id}",
+            order_id,
+            { order_id: order_id },
```
Comment (mrogers-5s) [inline lib/client/order.rb]: Drop the `order_id` in the data object, and add the BC errors.

File: lib/client/order.rb [hunk 6]
```diff
@@ -54,7 +75,14 @@ def get_shipments(order_id, params = {})
           response = client.get(uri({ order_id: order_id }, 'shipments'), params)
           return response.body
         rescue Faraday::Error => e
-          raise BigcommerceApiError.new('get order shipments', { order_id: order_id }, e)
+          raise BigcommerceApiError.new(
+            e.response[:status],
+            'get order shipments',
+            "Failed to get order shipments #{order_id}",
+            order_id,
+            { order_id: order_id },
```
Comment (mrogers-5s) [inline lib/client/order.rb]: Drop the `order_id` in the data object, and add the BC errors.

File: lib/client/order.rb [hunk 7]
```diff
@@ -64,7 +92,13 @@ def get_coupons(order_id, params = {})
           response = client.get(uri({ order_id: order_id }, 'coupons'), params)
           return response.body
         rescue Faraday::Error => e
-          raise BigcommerceApiError.new('get order coupons', { order_id: order_id }, e
+          raise BigcommerceApiError.new(
+            e.response[:status],
+            'get order coupons',
+            "Failed to get order coupons #{order_id}",
+            order_id,
+            { order_id: order_id },
```
Comment (mrogers-5s) [inline lib/client/order.rb]: Drop the `order_id` in the data object, and add the BC errors.

File: lib/client/order.rb [hunk 8]
```diff
@@ -75,7 +109,14 @@ def get_transactions(order_id, params = {})
           response = client.get(uri({ api_version: 'v3', order_id: order_id }, 'transactions'), params)
           return response.body['data']
         rescue Faraday::Error => e
-          raise BigcommerceApiError.new('get order transactions', { order_id: order_id }, e)
+          raise BigcommerceApiError.new(
+            e.response[:status],
+            'get order transactions',
+            "Failed to get order transactions #{order_id}",
+            order_id,
+            { order_id: order_id },
```
Comment (mrogers-5s) [inline lib/client/order.rb]: Drop the `order_id` in the data object, and add the BC errors.

File: lib/huginn_bigcommerce_order_agent/bigcommerce_order_agent.rb [hunk 9]
```diff
@@ -153,6 +153,7 @@ def handle(event)
         create_event payload: new_event.merge(
           status: faraday_error.response[:status],
           scope: e.scope,
+          message: e.message,
           response: faraday_error.response[:body],
           request_data: e.data,
```
Comment (mrogers-5s) [inline lib/huginn_bigcommerce_order_agent/bigcommerce_order_agent.rb]: This needs to create an event payload from the API error instance we created. Look at how the product agent is doing it and have this do the same thing.

**Review by mrogers-5s — CHANGES_REQUESTED** [review]

File: lib/huginn_bigcommerce_order_agent/bigcommerce_order_agent.rb (hunk 9 above)
Comment (jhaltercmu2020) [reply lib/huginn_bigcommerce_order_agent/bigcommerce_order_agent.rb]: @mrogers-5s Like this?

File: lib/client/order.rb (hunk 8 above)
Comment (jhaltercmu2020) [reply lib/client/order.rb]: @mrogers-5s resolved

File: lib/client/order.rb (hunk 7 above)
Comment (jhaltercmu2020) [reply lib/client/order.rb]: @mrogers-5s resolved

File: lib/client/order.rb (hunk 6 above)
Comment (jhaltercmu2020) [reply lib/client/order.rb]: @mrogers-5s resolved

File: lib/client/order.rb (hunk 5 above)
Comment (jhaltercmu2020) [reply lib/client/order.rb]: @mrogers-5s resolved

File: lib/client/order.rb (hunk 4 above)
Comment (jhaltercmu2020) [reply lib/client/order.rb]: @mrogers-5s resolved

File: lib/client/order.rb (hunk 3 above)
Comment (jhaltercmu2020) [reply lib/client/order.rb]: @mrogers-5s resolved

File: lib/client/customer.rb (hunk 2 above)
Comment (jhaltercmu2020) [reply lib/client/customer.rb]: @mrogers-5s resolved

**Review by mrogers-5s — APPROVED** [review]

**Comment by jhaltercmu2020** [comment]
@Ken-Michalak Ready for review

**Review by Ken-Michalak — APPROVED** [review]

**Pull request merged**